### ISSUE TITLE
Builds for ocaml < 4.03 with opam2

### ISSUE THREAD

**Issue by NathanReb** [ocaml/ocaml-ci-scripts#241]
It seems like using `.travis-docker.sh` and opam2 it's not currently possible to build with ocaml verions older than `4.03`.

I had a `4.02.3` build for [key-parsers](https://github.com/cryptosense/key-parsers) but there's no pre-installed 4.02 switch as part of the `ocaml/opam2:debian-stable` image as stated by the error:

```
Step 6/13 : RUN opam switch 4.02.3
 ---> Running in 7bf160a26fad
[ERROR] No switch 4.02.3 is currently installed. Did you mean 'opam switch create 4.02.3'?
        Installed switches are:
          - 4.07
          - 4.06
          - 4.05
          - 4.04
          - 4.03
```

Would it be possible to add such a switch? If not do you have any suggestions besides forcing `opam 1.2.2`?

Thanks!

**Comment by samoht** [comment]
The 4.03, ..., 4.07 are built in the Docker images, so switching to them is fast. I guess we could add a fallback in the Docker image to try to create a new switch with old compiler names instead of failing, e.g. in your case generate an `opam switch create 4.02.3`.

The line to change is https://github.com/ocaml/ocaml-ci-scripts/blob/master/.travis-docker.sh#L58 if you feel like giving the patch a go.

**Comment by NathanReb** [comment]
Wouldn't it be possible to add a 4.02 switch to the docker image? Creating a new switch is probably going to take ages so at this point I'd rather drop support for Ocaml < 4.03 although I agree it would still be better than nothing!

I'll take a look for the patch!

**Comment by avsm** [comment]
I could add a 4.02 switch, but the compiler support matrix is getting huge... and the disk space usage along with it for the container images.  I would recommend the manual switch compilation, or gently letting 4.02.3 go into that unsupported night.

**Comment by NathanReb** [comment]
Fixed by #242
